test/lib/eventually.hh

```diff
@@ -49,5 +51,16 @@ bool eventually_true(noncopyable_function<bool ()> f) {
     return false;
 }
 
-#define REQUIRE_EVENTUALLY_EQUAL(a, b) BOOST_REQUIRE(eventually_true([&] { return a == b; }))
-#define CHECK_EVENTUALLY_EQUAL(a, b) BOOST_CHECK(eventually_true([&] { return a == b; }))
+// Must be called in a seastar thread
+template <typename T>
+void REQUIRE_EVENTUALLY_EQUAL(std::function<T()> a, T b) {
+    eventually_true([&] { return a() == b; });
+    BOOST_REQUIRE_EQUAL(a(), b);
```

Comment:
This creates an opportunity for a confusing test failure, where `eventually_true()` fails, but the `BOOST_REQUIRE_EQUAL` succeeds here. I think it has a low enough chance of happening that we can let it slip.

Reply:
The outcome is still equivalent to eventually_true succeeding (eventually)